envi/tests/test_arch_h8.py

```diff
@@ -0,0 +1,1084 @@
+        if len(emu.curpath[2]['readlog']):
+            outstr = emu.curpath[2]['readlog']
+            if len(outstr) > 10000: outstr = outstr[:10000]
+            print( repr(op) + '\t\tRead: ' + repr(outstr) )
+        if len(emu.curpath[2]['writelog']):
+            outstr = emu.curpath[2]['writelog']
+            if len(outstr) > 10000: outstr = outstr[:10000]
+            print( repr(op) + '\t\tWrite: '+ repr(outstr) )
+        emu.curpath[2]['readlog'] = []
+        emu.curpath[2]['writelog'] = []
+
+        return not success
+
+    def test_parsers_rudimentary(self, buf = 'ABCDEFGHIJKLMNOP', off=3, va=0x2544):
+        val, = struct.unpack_from('>H', buf, off)
+
+        for tsize in (1,2,4):
+            p_i3_Rd(va, val, buf, off, tsize)
+            p_i3_aERd(va, val, buf, off, tsize) 
+            p_i3_aAA8(va, val, buf, off, tsize) 
+            p_i8_CCR(va, val, buf, off, tsize) 
+            p_i8_Rd(va, val, buf, off, tsize) 
+            p_i16_Rd(va, val, buf, off, tsize) 
+            p_i32_ERd(va, val, buf, off, tsize) 
+            p_Rd(va, val, buf, off, tsize) 
+            p_Rs_Rd(va, val, buf, off, tsize)  
+            p_Rs_Rd_4b(va, val, buf, off, tsize)  
+            p_Rs_ERd(va, val, buf, off, tsize)  
+            p_Rs_ERd_4b(va, val, buf, off, tsize)  
+            p_ERd(va, val, buf, off, tsize)  
+            p_ERs_ERd(va, val, buf, off, tsize)  
+            p_Rn_Rd(va, val, buf, off, tsize)  
+            p_Rn_aERd(va, val, buf, off, tsize)  
+            p_Rn_aAA8(va, val, buf, off, tsize)  
+            p_aERn(va, val, buf, off, tsize)  
+            p_aAA24(va, val, buf, off, tsize)  
+            p_aaAA8(va, val, buf, off, tsize)  
+            p_disp8(va, val, buf, off, tsize)  
+            p_disp16(va, val, buf, off, tsize)  
+            p_nooperands(va, val, buf, off, tsize) 
+
+        h8m = envi.getArchModule('h8')      
+
+        for instr in raw_instrs:
+            inst = instr[0]
+            op = h8m.archParseOpcode(inst, 0, 0x50)
+            print( "%26s %s" % (instr[0].encode('hex'), op) )
+            if len(op) != len(inst):
+                #raise Exception(" LENGTH FAILURE:  expected: %d  real: %d  '%s'" % (len(inst), len(op), inst.encode('hex')))
+                print(" LENGTH FAILURE:  expected: %d  real: %d  '%s'" % (len(inst), len(op), inst.encode('hex')))
+
+
+
+def generateTestInfo(ophexbytez='6e'):
+    h8 = e_h8.H8Module()
+    opbytez = ophexbytez
+    op = h8.archParseOpcode(opbytez.decode('hex'), 0, 0x4000)
+    print( "opbytez = '%s'\noprepr = '%s'"%(opbytez,repr(op)) )
+    opvars=vars(op)
+    opers = opvars.pop('opers')
+    print( "opcheck = ",repr(opvars) )
+
+    opersvars = []
+    for x in range(len(opers)):
+        opervars = vars(opers[x])
+        opervars.pop('_dis_regctx')
+        opersvars.append(opervars)
+
+    print( "opercheck = %s" % (repr(opersvars)) )
+
+
+
+raw_instrs = [
+    ('8340'.decode('hex'), ),
+    ('0832'.decode('hex'), ),
```

Comment:
seems like the convention for these would be smaller/cleaner if they were all declared, then decode('hex') done in a loop :D